src/pages/index.js

```diff
@@ -0,0 +1,372 @@
+import { enableValidation, validationConfig } from "../scripts/validation.js";
+import "../pages/index.css";
+import Api from "../utils/Api.js";
+import { setButtonText } from "../utils/helpers.js";
+
+// Declare cardTemplate before using it
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "54273f74-b0c9-4977-a451-bfc48ef2742e",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getAppInfo()
+  .then(([cards, userData]) => {
+    if (Array.isArray(cards) && userData) {
+      profileName.textContent = userData.name;
+      profileDescription.textContent = userData.about;
+      profileAvatar.src = userData.avatar;
+
+      cards.forEach((item) => {
+        const cardElement = getCardElement(item);
+        cardsList.append(cardElement);
+      });
+    } else {
+      throw new Error("Invalid data format");
+    }
+  })
+  .catch((error) => {
+    console.error("Failed to fetch app info:", error);
+  });
+
+// profile elements
+const profileEditButton = document.querySelector(".profile__edit-button");
+const cardModalButton = document.querySelector(".profile__add-button");
+const profileName = document.querySelector(".profile__name");
+const profileDescription = document.querySelector(".profile__description");
+const profileAvatar = document.querySelector(".profile__avatar");
+const avatarModalButton = document.querySelector(".profile__avatar-btn");
+
+// Edit form elements
+const editModal = document.querySelector("#edit-profile-modal");
+const editFormElement = editModal.querySelector(".modal__form");
```

Comment:
**COULD BE IMPROVED**

We can get any form using their **unique** attribute `name` or `id` from `document.forms`

```js

const profileForm = document.forms["profile-form"];
const cardForm = document.forms["card-form"];

```

You need to add `name` or `id` first in `index.html` to achieve that.

More info here https://developer.mozilla.org/en-US/docs/Web/API/Document/forms